kfdef/source/README.md

```diff
@@ -1,6 +1,8 @@
 This directory contains kustomization packages that are used to generate the YAML specs for the KFDef.
 
 The script `hack/build_kfdef_specs.py` is used to run kustomize and output the YAML files.
+By default *metadata.name* is not generated. However, `OPERATOR_KFDEF` environment variable can be
+set to `true` for building the KubeFlow operator version of kfdefs that contain *metadata.name*.
 
```

Comment:
Would it cause any harm if *metadata.name* is generated for every kfdef?

Reply:
https://github.com/kubeflow/manifests/blob/15ac5f96173375b7df04243ab6424c840fc1be2f/hack/build_kfdef_specs.py#L60-L62
Based on the comments it preferred to let kfctl to fill in the name instead of generating it with the script.

Reply:
@jlewi @kunmingg 

what's the rationale to omit the *metadata.name* from kfdef? Previously the operator could directly use them, and now we have to tell user to do manipulation in order to be able to use them

Reply:
Yes; generating a name for every KFDef will be a significant behavior change.